README.md

```diff
@@ -59,6 +59,41 @@ val marker = googleMap.addMarker {
 }
 ```
 
+#### Coroutines
+
+Accessing a `GoogleMap` instance can be retrieved using coroutines vs. traditional the callback mechanism. The example here demonstrates how you can use this feature alongside with [Lifecycle-aware coroutine scopes][lifecycle] provided in Android’s Architecture Components.
+
+**NOTE**: This is an experimental feature and can only be used by using the `MapsExperimentalFeature` annotation class.
+
+_Before_
+```java
+@Override
+public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    SupportMapFragment mapFragment = ((SupportMapFragment) getSupportFragmentManager().findFragmentById(R.id.map));
+
+    mapFragment.getMapAsync(new OnMapReadyCallback {
+        @Override
+        public void onMapReady(GoogleMap map) {
+            // Access GoogleMap instance here
+        }
+    });
+}
+```
+
+_After_
+```kotlin
+@MapsExperimentalFeature
+override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as! SupportMapFragment
+
+    lifecycle.coroutineScope.launchWhenCreated {
+        val googleMap = mapFragment.awaitMap()
+    }
+}
+```
+
```

Comment:
It looks like GitHub won't let me add a suggestion that includes a code block (?), but I'd suggest something like this here:

```suggestion

Note that to use the [Lifecycle-aware coroutine scope][lifecycle] you'll need to add the following to your `build.gradle` dependencies:

`implementation 'androidx.lifecycle:lifecycle-runtime-ktx:<latest-version>'`

```